<launch>
    <arg name="robot_name" default="lwr"/>
    <arg name="base_frame"  default="link_0"/>
    <arg name="urdf_url"  default="$(find orca)/examples/lwr.urdf"/>

    <node name="gazebo_node" pkg="orca_ros" type="gazebo_node" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="urdf_url" value="$(arg urdf_url)"/>
    </node>

    <node name="minimal_controller" pkg="orca_ros" type="minimal_controller" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="urdf_url" value="$(arg urdf_url)"/>
        <param name="controller_name" value="$(arg robot_name)_orca_controller"/>
    </node>

    <node name="minimal_client" pkg="orca_ros" type="minimal_client" output="screen">
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="controller_name" value="$(arg robot_name)_orca_controller"/>
    </node>

    <group ns="/orca">
        <group ns="$(arg robot_name)">

            <param name="robot_description" command="$(find xacro)/xacro '$(arg urdf_url)' --inorder"/>
            <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
        </group>
    </group>
</launch>
